src/basicfuns.jl

```diff
@@ -165,9 +165,14 @@ Return `log(1+exp(x))` evaluated carefully for largish `x`.
 This is also called the ["softplus"](https://en.wikipedia.org/wiki/Rectifier_(neural_networks))
 transformation, being a smooth approximation to `max(0,x)`. Its inverse is [`logexpm1`](@ref).
 
+The generalized `softplus` function (Wiemann et al., 2024) takes an additional optional parameter `a` that control 
```

Comment:
I assume there exist earlier references for this function?

Reply:
I went through [Liu and Farber](https://[arxiv.org/pdf/1706.03609](https://arxiv.org/pdf/1706.03609)) to double-check

From my understanding (ML is not my  field), they validate "noisy softplus" as an improvement over other activation functions for neurons in NNs. 

![image](https://github.com/user-attachments/assets/9b1bbf12-c8a4-4aa0-b7b7-c0788a99c9ef)

However, it seems like they  named the `a` parameter sigma. Their plot looks similar but different in terms of values (?)

![image](https://github.com/user-attachments/assets/956d79c0-74bf-4c9f-a687-ab5faced4a68)

> I'm not sure how widely use this variant 

I share your concern here, I'm also careful not to add niche features to such a base package and add maintaining burden. 
I can't say how commonly the generalized version is already used, its development  seems fairly recent.
However, I can see its usefulness in quite a lot of cases: the default softplus only becomes close to identity after x > 2, and from experience we often do model parameters smaller than that (typical sigmas in my neuroscience/psychology are like between 0 and 1), so using adjust softplus links would make sense in these contexts. I suppose it's a tradeoff between the complexity of the feature and its (potential) usage